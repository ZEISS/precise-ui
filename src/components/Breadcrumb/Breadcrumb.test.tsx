import * as React from 'react';
import * as enzyme from 'enzyme';
import { Breadcrumb } from './';

describe('<Breadcrumb />', () => {
  it('should render a basic <Breadcrumb>', () => {
    const wrapper = enzyme.shallow(<Breadcrumb title="Title" to="/test" />);
    expect(wrapper.text()).toBe('<Styled(Anchor) />');
  });

  it('should render a <Breadcrumb> with external reference', () => {
    const wrapper = enzyme.shallow(<Breadcrumb title="Title" href="foo" />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should render a <Breadcrumb> with internal reference', () => {
    const wrapper = enzyme.shallow(<Breadcrumb title="Title" to="foo" />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should execute the click handler', () => {
    let clicked = false;
    const clickHandler = () => {
      clicked = true;
    };
    const wrapper = enzyme.shallow(<Breadcrumb title="Title" onClick={clickHandler} />);
    wrapper.simulate('click', {
      preventDefault() {},
    });
    expect(clicked).toBe(true);
  });

  it('should not fail if no history given', () => {
    let clicked = false;
    const clickHandler = () => {
      clicked = true;
    };
    const wrapper = enzyme.shallow(<Breadcrumb title="Title" to="/foo" onClick={clickHandler} />);
    wrapper.simulate('click', {
      preventDefault() {},
    });
    expect(clicked).toBe(true);
  });
});
