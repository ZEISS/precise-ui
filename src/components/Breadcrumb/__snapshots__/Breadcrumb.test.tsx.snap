// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Breadcrumb /> should render a <Breadcrumb> with external reference 1`] = `
<Styled(Anchor)
  href="foo"
  title="Title"
>
  Title
</Styled(Anchor)>
`;

exports[`<Breadcrumb /> should render a <Breadcrumb> with internal reference 1`] = `
<Styled(Anchor)
  title="Title"
  to="foo"
>
  Title
</Styled(Anchor)>
`;
